controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/gui/GUIHandler.java

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.restapi.gui;
+
+import com.google.inject.Inject;
+import com.yahoo.container.jdisc.HttpRequest;
+import com.yahoo.container.jdisc.HttpResponse;
+import com.yahoo.container.jdisc.LoggingRequestHandler;
+import com.yahoo.vespa.hosted.controller.restapi.ErrorResponse;
+import com.yahoo.yolean.Exceptions;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Level;
+
+
+public class GUIHandler extends LoggingRequestHandler {
+
+    @Inject
+    public GUIHandler(Context parentCtx) {
+        super(parentCtx);
+    }
+
+    @Override
+    public HttpResponse handle(HttpRequest request) {
+        try {
+            switch (request.getMethod()) {
+                case GET: return handleGET(request);
+                default: return ErrorResponse.methodNotAllowed("Method '" + request.getMethod() + "' is not supported");
+            }
+        } catch (IllegalArgumentException e) {
+            return ErrorResponse.badRequest(Exceptions.toMessageString(e));
+        } catch (RuntimeException e) {
+            log.log(Level.WARNING, "Unexpected error handling '" + request.getUri() + "'", e);
+            return ErrorResponse.internalServerError(Exceptions.toMessageString(e));
+        }
+    }
+
+    private HttpResponse handleGET(HttpRequest request) {
+        com.yahoo.vespa.hosted.controller.restapi.Path path = new com.yahoo.vespa.hosted.controller.restapi.Path(request.getUri().getPath());
```

Comment:
You cannot have this dependency - if you want to reuse Path you can move it to the vespajlib module.

Reply:
Yes. Maybe create a new restapi package under vespajlib.

Reply:
I had to move `ConfigServerException` from controller-api and `ErrorResponse`, `Path` and `SlimeJsonResponse` from controller-server to container-core under a new restapi package

Reply:
Have problem importing these into the test. How can I be able to use these? 
```
import com.yahoo.application.Networking;
import com.yahoo.application.container.JDisc;
import com.yahoo.application.container.handler.Request;
import com.yahoo.application.container.handler.Response;
```

Reply:
Heh ... you need to find another solution for ConfigServerException, that doesn't make any sense in container-core.

Reply:
You cannot depend on application from container-search because there is a necessary dependency in the other direction. Let's discuss that tomorrow/tonight, I have to run now.

Reply:
Ah.. Okey. Talk tomorrow then.

Reply:
Ah.. Okey. We'll talk later!

Reply:
These imports in the GUIHandlerTest.java, are as well inaccessible from container-search..

```
import com.yahoo.application.Networking;
import com.yahoo.application.container.JDisc;
import com.yahoo.application.container.handler.Request;
import com.yahoo.application.container.handler.Response;
```

Reply:
Moved the com.yahoo.vespa.hosted.provision.restapi.v2.ErrorResponse to Container-Core, instead of the one I used in the first place. It did not use ConfigServerException, and worked with GUIHandler, I believe. Will moving this class cause any problems for other classes?

Reply:
Please try to avoid moving anything in `controller-api ` as that package declares the interface between internal code and `controller-server` and may break if changed.

Reply:
Yes, that won't work. I suggest you give up reusing any of the response classes. If you leave those where they were I think you can move just Path to vespajlib.

Reply:
That leaves just one problem: You can't have tests that uses application in container-search. I suggest you make a new module container-search-gui which depends on container-search and application.